### PR TITLE
change default values of Adaboost

### PR DESCRIPTION
**Pull request by ViswanathBalla22**
The default values for the Adaboost training options have been updated to enhance the algorithm's performance. This update includes changes to the maximum number of boosting iterations, tolerance for convergence, and default weak learner type. These adjustments aim to improve the out-of-the-box performance of the Adaboost algorithm.

Related issue: #3010

### CODE REVIEW

**Review by rcurtin — CHANGES_REQUESTED** [review]
@ViswanathBalla22 I don't understand why you have chosen the values that you have here.  There doesn't appear to be any reason for them.  If you specifically think that 226 is the best default number of iterations for AdaBoost, please explain why.  If we are going to choose better default values, we need to validate their performance on a variety of datasets.  A ChatGPT-generated PR description is not sufficient.

**Comment by ViswanathBalla22** [comment]
hi @rcurtin @zoq , I  want to express my sincere apologies if I've made any mistakes,
After reviewing this discussion [https://stats.stackexchange.com/questions/163560/how-many-adaboost-iterations], I observed that even a lower number of iterations (226) is sufficient for Adaboost. More iterations don’t always significantly improve performance, and limiting them helps prevent overfitting. So  I have  decided to adjust the default value of iterations in Adaboost from 1,000 to 226. Firstly, by limiting the iterations, we can stop the algorithm early if it's performing well before reaching the maximum. This helps save time and resources. Secondly, fewer iterations reduce the risk of overfitting, where the model fits too closely to the training data and does not generalize well to new data.

**Comment by ViswanathBalla22** [comment]
@rcurtin ,  I propose 226 iterations. As Adaboost changes were mistakenly included in the MDL penalty PR, if you  approved, I'll open a new PR specifically for Adaboost adjustments, otherwise, I'll close this PR.

File: src/mlpack/methods/decision_tree/decision_tree.hpp
Comment (conradsnicta) [inline src/mlpack/methods/decision_tree/decision_tree.hpp:45]: what does adding MDLPenaltyType have to do with changing default adaboost values?

**Comment by conradsnicta** [comment]
> I propose 226 iterations.

Why 226 and not 227, or 225, or 10, or 500?

The mentioned discussion on stackexchange doesn't seem to advocate for 226.

Please open a fresh PR with clear justifications (including empirical evidence on several datasets) for the proposed default values.  Also, ensure that the PR only modifies adaboost related files.

**Comment by rcurtin** [comment]
For what it's worth, I simply won't approve a PR unless you can empirically demonstrate on several datasets that the new defaults give better performance than the existing settings.

**Comment by ViswanathBalla22** [comment]
@rcurtin @conradsnicta @zoq 
I completely understand. I'll make sure to conduct proper tests and research before submitting any future PRs. As I'm relatively new to open source contributions, I appreciate your patience and guidance in helping me improve. Thank you for your support!

**Comment by rcurtin** [comment]
@ViswanathBalla22 sounds good.  It's okay that you use ChatGPT to communicate, but it is expected that you fully understand the code that you are submitting---otherwise you are just wasting maintainers' time.  LLMs generally do not produce code that is ready for a PR, and it has to be manually checked over, built locally, tested, and understood before a PR is opened.  I'm not interested in spending my time doing that until you have done it first. :+1:

**Pull request closed without merging**
